datafusion/src/physical_plan/file_format/parquet.rs

```diff
@@ -473,6 +536,69 @@ mod tests {
         schema::types::SchemaDescPtr,
     };
 
+    #[tokio::test]
```

Comment:
Thanks! This is a great start on testing 

I was thinking there is no reason to use checked in parquet files for this test -- we can create files as part of the test. I went ahead and coded this up as https://github.com/coralogix/arrow-datafusion/pull/1 as the scaffolding was a bit annoying. 

With that code I suggest we test:
1. columns in different orders in different files (e.g. one file has (`a`, `b`, `c`) columns and one has (`b`, `a`)
2. projection (which I do see is covered here)
3.  Column with the same name and different types
4. Columns with different subsets of the data  (`a`, `b`) and (`b`, `c`) for example

Reply:
On point 1, I actually noticed this morning that my implementation would fail in the case where the columns are in different orders but all projected columns are present in the file. The easiest way to fix that would be to remove the condition on re-mapping the columns in the output batch (so do that mapping in all cases). 

Are we concerned about the runtime cost of that operation and try to avoid it if unnecessary? I'm relatively new to Rust so not sure how expensive cloning `Arc` is.

Reply:
>  I'm relatively new to Rust so not sure how expensive cloning Arc is.

`Clone`ing an Arc is very fast (it increments an atomic counter) 🚤